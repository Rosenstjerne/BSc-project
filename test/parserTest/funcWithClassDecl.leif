class Person {
  var int name;
  var int age;
}

function int greet(Person p) {
  return p.name + p.age;
}

var Person john;
john = new Person;
john.name = 1;
john.age = 30;

print(greet(john));
