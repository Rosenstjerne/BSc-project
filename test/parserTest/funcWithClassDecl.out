function null main() {
	class Person {
		var int name;
		var int age;
		}
	
	var Person john;
	
	function int greet(Person p) {
		return ((p.name) + (p.age));
		
	}
	
	john = (new Person);
	(john.name) = 1;
	(john.age) = 30;
	print(greet(john));
	
}
